### PR TITLE
Feature: Add The Ability To Hide Viewall Links

### PR DESCRIPTION
**Pull request by sghoweri**
## Summary
Adds the ability to hide any viewall links that would normally show up in the Pattern Lab navigation.

## Details
Updates the Webpack build config + `.patternlabrc` config to support a new `noViewAll` config option when customizing Pattern Lab's UI.

### Before vs After
image

image

## How To Test

1. Pull down this branch and install NPM dependencies by running `npm run setup`.
2. Go to the `packages/edition-twig` folder and uncomment line 5 in the example `.patternlabrc` config file.

```
module.exports = {
  buildDir: __dirname + '/public',
  wwwDir: __dirname + '/public/',
  publicPath: '/public/styleguide/',
  // noViewAll: true, // <!-- uncomment this line
};
```
3. Rebuild Pattern Lab's UI by running `npm run build:uikit` from the same `packages/edition-twig` folder
4. Confirm the regenerated UIKit is rendering properly by now running `npm run serve`

image

### CODE REVIEW

**Comment by jryanconklin** [comment]
Just a note to say I'd really love to have this.

**Comment by bmuenzenmeyer** [comment]
@sghoweri can i switch this target to `dev` ?

**Comment by sghoweri** [comment]
@bmuenzenmeyer go for it!

File: packages/uikit-workshop/.patternlabrc.example.js [hunk 1]
```diff
@@ -2,4 +2,5 @@
 
 module.exports = {
   // buildDir: __dirname + '/www/pattern-lab',
+  noViewAll: true, // uncomment to disable displaying viewAll links in Pattern Lab
```
Comment (bmuenzenmeyer) [inline packages/uikit-workshop/.patternlabrc.example.js]: this comment doesnt make sense, as it's enabled, not commented

Comment (sghoweri) [reply packages/uikit-workshop/.patternlabrc.example.js]: Derp — I think I was thinking we’d want this option to be disabled by default (hence the comment) but I probably I uncommented that when I was putting together my PR screenshot 🤦‍♂️

**Comment by sghoweri** [comment]
> playing catchup on this feature - it is not accessible from a `patternlab-config.json` entry, correct? ony at build time from the uikit?
> 
> EDIT:
> 
> looking at that edition for the first time - adding `build:uikit` is an interesting end-user customization feature! 👍
> 
> $INIT_CWD is not cross-platform safe - going to try cross-env

@bmuenzenmeyer yeah customizing / extending PL is an exciting area I’m hoping to spend some time planning out / documenting / building up!

At a high level, I’m thinking about 3 overall ways PL’s UI could get customized:
1. Via runtime / JSON config (`patternlab-config.json`) not requiring the UI to recompile
2. Via `patternlabrc.js` (requiring UI to get rebuilt, say via the CLI)
3. And declaratively via customizing the web component HTML getting used

File: packages/uikit-workshop/.patternlabrc.example.js (hunk 1 above)
Comment (bmuenzenmeyer) [reply packages/uikit-workshop/.patternlabrc.example.js]: bump

**Comment by bmuenzenmeyer** [comment]
@sghoweri can you change this to use cross-env? otherwise I can try quick

it'd be nice to batch this in to the release (I hope we release today)

**Comment by sghoweri** [comment]
@bmuenzenmeyer I can check this out now - no prob

**Comment by bmuenzenmeyer** [comment]
I am going to try to get this working on Windows then.

**Comment by bmuenzenmeyer** [comment]
Also like to get this into `development-edition-engine-handlebars` for now

**Comment by bmuenzenmeyer** [comment]
trying to work through this

```
Error: ENOENT: no such file or directory, open 'D:\src\patternlab-node\packages\development-edition-engine-handlebars\node_modules\@pattern-lab\uikit-workshop\$INIT_CWD\.patternlabrc.js'
```

**Comment by bmuenzenmeyer** [comment]
this works!?

````
"build:uikit": "npm run build --prefix node_modules/@pattern-lab/uikit-workshop -- --patternlabrc ../../../.patternlabrc.js",
```

**Comment by bmuenzenmeyer** [comment]
I am still seeing the pattern type view all, but sub type appears gone

image

does this seem right @sghoweri

**Comment by sghoweri** [comment]
Hmmm

**Comment by bmuenzenmeyer** [comment]
changing noViewAll seems to have no effect for me within my codebase

**Comment by bmuenzenmeyer** [comment]
pushed https://github.com/pattern-lab/patternlab-node/tree/1034-test

```
npm run setup
cd packages/development-edition-engine-handlebars
change `noViewAll` within `.patternlabrc.js`
npm run build:uikit
npm run pl:serve
```

**Comment by bmuenzenmeyer** [comment]
@sghoweri game to release `dev` as is and figure this out as a fast-follower?

**Comment by sghoweri** [comment]
> @sghoweri game to release `dev` as is and figure this out as a fast-follower?

@bmuenzenmeyer I'm good with that!

**Comment by bmuenzenmeyer** [comment]
@sghoweri just checking in on this - very excited for the functionality but i want to ensure it works in both Twig and Handlebars editions before we move forward. will try again after a merge sometime soon

**Comment by bmuenzenmeyer** [comment]
once this merge conflict is resolved, I am hoping that #1098 will immensely help iterate on this solution

**Comment by bmuenzenmeyer** [comment]
tracking this work in https://github.com/pattern-lab/patternlab-node/tree/sghoweri-feature/no-viewall-links

**Comment by bmuenzenmeyer** [comment]
i've spent a fair amount of time with this and cannot get it to work - i wonder now that we have the `build-tools.js` file if parts of that could be used. 

@sghoweri you've also expressed concern that this might not be the best approach. thoughts on making this a runtime config option?

**Comment by sghoweri** [comment]
> i've spent a fair amount of time with this and cannot get it to work - i wonder now that we have the `build-tools.js` file if parts of that could be used.
> 
> @sghoweri you've also expressed concern that this might not be the best approach. thoughts on making this a runtime config option?

@bmuenzenmeyer no worries, appreciate you taking a look at this.

I'm looking this over this morning (probably trying to switch to using a runtime config out of simplicity); I'll see what I can do!

**Comment by bmuenzenmeyer** [comment]
It's entirely possible I'm either doing something wrong (though I was porting the twig edition code directly) or the is a simple cross platform fix... but I haven't found it yet.

Thanks for plugging away on this with me!

**Comment by sghoweri** [comment]
@bmuenzenmeyer this should be good to go! Updated the PR description + adjusted some stuff internally to now have this be based off of the runtime config vs having to recompile the UI in order to enable / disable this.

**Comment by bmuenzenmeyer** [comment]
will give this a go later this weekend @sghoweri - thanks!

File: packages/edition-twig/.patternlabrc.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-  buildDir: __dirname + '/public',
-  wwwDir: __dirname + '/public/',
-  publicPath: '/public/styleguide/',
+  // target the UIKit installed / symlinked under node_modules
+  buildDir: __dirname + '/node_modules/@pattern-lab/uikit-workshop/dist',
+  // noViewAll: true,
```
Comment (bmuenzenmeyer) [inline packages/edition-twig/.patternlabrc.js]: I think this can be removed

**Comment by bmuenzenmeyer** [comment]
> I don't think there is any patternlab.io documentation on the `theme` config.

I was wrong - we did add it here https://patternlab.io/docs/advanced-config-options.html but it could certainly be improved with our rewrite / refresh

**Pull request merged**